llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -1079,6 +1079,12 @@ static void simplifyRecipe(VPRecipeBase &R, VPTypeAnalysis &TypeInfo) {
     return;
   }
 
+  if (match(Def, m_Select(m_True(), m_VPValue(X), m_VPValue())))
+    return Def->replaceAllUsesWith(X);
+
+  if (match(Def, m_Select(m_False(), m_VPValue(), m_VPValue(X))))
+    return Def->replaceAllUsesWith(X);
```

Comment:
The fold on false isn't immediately used by the test cases in tree, but is needed in an upcoming patch, so I included it with the fold on true